webapp/src/ts/services/translate-locale.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { map, shareReplay, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TranslateLocaleService {
+  constructor(
+    private translateService:TranslateService,
+  ) {
+  }
+
+  private loadingTranslations = {};
+
+  private getTranslation(locale) {
+    if (this.translateService.translations[locale]) {
+      return;
+    }
+
+    if (this.loadingTranslations[locale]) {
+      return;
+    }
+
+    const loadingTranslations = this.translateService.currentLoader.getTranslation(locale).pipe(shareReplay(1), take(1));
+    const translationsCompiled = loadingTranslations.pipe(map(((res) => this.translateService.compiler.compileTranslations(res, locale))), shareReplay(1), take(1));
```

Comment:
I find a bit hard to read this one with pipes and maps, I need to track where each parenthesis is opening/closing, so maybe putting then in separate lines and with indentation can improve the readability.

Reply:
Yea, definitely. I actually copied these word for word from the `ngx-translator` source code :D